zarr/_storage/store.py

```diff
@@ -168,3 +160,11 @@ def _listdir_from_keys(store: BaseStore, path: Optional[str] = None) -> List[str
             child = suffix.split('/')[0]
             children.add(child)
     return sorted(children)
+
+
+def _rmdir_from_keys(store: Union[BaseStore, MutableMapping], path: Optional[str] = None) -> None:
```

Comment:
Why did this need to shift down?

Reply:
It's just the order used everywhere else in the code: 1) listdir  2) rmdir